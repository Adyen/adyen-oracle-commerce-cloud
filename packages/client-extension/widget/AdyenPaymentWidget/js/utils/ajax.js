import $ from 'jquery'
import { baseAPIUrl, channels } from '../constants'

export default (isPreview = true, siteId) => {
    return (path, cb, method = 'get') => {
        $.ajax({
            url: `${baseAPIUrl}/${path}`,
            method,
            headers: {
                channel: isPreview ? channels.preview : channels.storefront,
                'x-ccsite': siteId
            },
            success: cb,
        })
    }
}
