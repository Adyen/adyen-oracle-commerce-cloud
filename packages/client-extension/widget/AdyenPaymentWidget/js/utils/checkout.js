import ccConstants from 'ccConstants'
import { store } from '../components'
import * as constants from '../constants'
import { eventEmitter } from './index'

export const getDefaultConfig = () => {
    const environment = store.get(constants.environment)
    const locale = store.get(constants.locale)
    const originKey = store.get(constants.originKey)
    const paymentMethodsResponse = store.get(constants.paymentMethodsResponse)

    return {
        showPayButton: true,
        locale,
        environment: environment.toLowerCase(),
        originKey,
        paymentMethodsResponse,
    }
}

export const createFromAction = ({ action, selector, checkoutComponent }) => {
    checkoutComponent.createFromAction(action).mount(selector)
}

class Checkout {
    constructor(type) {
        this.type = type
        this.checkout = undefined
    }

    // eslint-disable-next-line no-undef
    getCheckout = configuration => this.checkout || new AdyenCheckout({ ...getDefaultConfig(), ...configuration })
    createCheckout = ({ configuration, selector, type, options = {} }, cb) => {
        const checkout = this.getCheckout(configuration)
        checkout.create(type, options).mount(selector)

        cb && cb(checkout)
    }

    createStoredCardCheckout = cb => {
        const configuration = { onChange: this.onChange(), onSubmit: this.onSubmit() }
        const checkout = this.getCheckout(configuration)
        eventEmitter.store.emit(constants.storedPaymentMethods, checkout.paymentMethodsResponse.storedPaymentMethods)
        checkout.paymentMethodsResponse.storedPaymentMethods.forEach(storedPaymentMethod => {
            const selector = `#adyen-stored_${storedPaymentMethod.id}-payment`
            checkout.create(constants.card, storedPaymentMethod).mount(selector)
        })

        cb && cb(checkout)
    }

    onSubmit = () => () => {
        const loader = document.querySelector(`.loader-wrapper`)
        loader && loader.classList.toggle('hide', false)

        const order = store.get(constants.order)
        eventEmitter.payment.emit(constants.setPayment, this.type)

        order().id(null)
        order().op(ccConstants.ORDER_OP_INITIATE)
        order().handlePlaceOrder()
    }

    onChange = options => (state, component) => {
        const paymentDetails = store.get(constants.paymentDetails)
        const isValid = component.isValid && typeof state.data === 'object'
        eventEmitter.store.emit(constants.isValid, isValid)

        const payload = { ...paymentDetails, [this.type]: { ...state.data, ...options } }

        eventEmitter.store.emit(constants.paymentDetails, payload)
    }
}

export default Checkout
