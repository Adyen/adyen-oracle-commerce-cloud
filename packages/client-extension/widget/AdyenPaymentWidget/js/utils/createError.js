import notifier from 'notifier'
import navigation from 'navigation'
import eventEmitter from './events'
import { destroySpinner } from './spinner'
import * as constants from '../constants'

const createError = (options, preserveUrl = false) => ({
    errorMessage = 'The payment is REFUSED.',
} = {}) => {
    eventEmitter.store.emit(constants.isSubmitting, false)
    'redirectLink' in options &&
        navigation.goTo(options.redirectLink, preserveUrl)
    destroySpinner()
    notifier.sendError('adyen', errorMessage, true)
}

export default createError
