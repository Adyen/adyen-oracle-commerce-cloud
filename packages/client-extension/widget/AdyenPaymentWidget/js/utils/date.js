export const addDays = days => new Date().setDate(new Date().getDate() + days)
