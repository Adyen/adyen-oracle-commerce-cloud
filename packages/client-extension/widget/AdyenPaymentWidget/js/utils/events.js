import $ from 'jquery'

class Emitter {
    store = {
        emit: (key, value) => $.Topic('store').publish(key, value),
        listen: cb => $.Topic('store').subscribe(cb),
    }

    _ev = suffix => ({
        emit: (event, ...args) => $.Topic(`${suffix}-${event}`).publish(...args),
        on: (event, cb) => $.Topic(`${suffix}-${event}`).subscribe(cb),
    })

    emit = (key, ...args) => $.Topic(key).publish(...args)

    payment = this._ev('payment')
    order = this._ev('order')
    component = this._ev('component')

    on = (event, cb) => $.Topic(event).subscribe(cb)
}

export default new Emitter()
