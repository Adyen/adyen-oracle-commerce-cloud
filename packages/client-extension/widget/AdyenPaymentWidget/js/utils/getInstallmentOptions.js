const creditCards = [
    { id: 1, name: 'Amex', code: 'amex' },
    // {id: "argencard", name: "Argencard"},
    // {id: "bcmc", name: "Bancontact/Mister Cash"},
    // {id: "bijcard", name: "de Bijenkorf Card"},
    // {id: "cabal", name: "Cabal"},
    // {id: "cartebancaire", name: "Carte Bancaires"},
    // {id: "codensa", name: "Codensa"},
    // {id: "cup", name: "China Union Pay"},
    // {id: "dankort", name: "Dankort"},
    // {id: "diners", name: "Diners Club"},
    // {id: "discover", name: "Discover"},
    { id: 2, name: 'ELO', code: 'elo' },
    // {id: "forbrugsforeningen", name: "Forbrugsforeningen"},
    { id: 4, name: 'HiperCard', code: 'hiper' },
    { id: 4, name: 'HiperCard', code: 'hipercard' },
    // {id: "jcb", name: "JCB"},
    // {id: "karenmillen", name: "Karen Millen GiftCard"},
    // {id: "laser", name: "Laser"},
    // {id: "maestro", name: "Maestro"},
    // {id: "maestrouk", name: "Maestro UK"},
    { id: 8, name: 'Mastercard', code: 'mc' },
    // {id: "mcalphabankbonus", name: "Alpha Bank Mastercard Bonus"},
    // {id: "mir", name: "MIR"},
    // {id: "naranja", name: "Naranja"},
    // {id: "oasis", name: "Oasis GiftCard"},
    // {id: "shopping", name: "Tarjeta Shopping"},
    // {id: "solo", name: "Solo"},
    // {id: "troy", name: "Troy"},
    // {id: "uatp", name: "UATP"},
    { id: 16, name: 'Visa', code: 'visa' },
    // {id: "visaalphabankbonus", name: "Alpha Bank Visa Bonus"},
    // {id: "visadankort", name: "Visa Dankort"},
    // {id: "warehouse", name: "Warehouse GiftCard"},
]

function getInstallmentOptions(installmentsOptions, amount, brand) {
    const creditCard = creditCards.find(({ code }) => brand === code)
    const isAllowedCreditCard = creditCardSum =>
        (parseInt(creditCardSum) & creditCard.id) !== 0
    const createInstallmentsObj = item => {
        const [amountRange, numberOfInstallments, creditCardSum] = item
        const isValid = creditCard && isAllowedCreditCard(creditCardSum)

        return isValid && amount >= amountRange && { numberOfInstallments }
    }

    return installmentsOptions
        .map(createInstallmentsObj, [])
        .filter(valid => valid)
}

export default getInstallmentOptions
