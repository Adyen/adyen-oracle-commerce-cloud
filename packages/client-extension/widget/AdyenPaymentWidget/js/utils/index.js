export { createSpinner, destroySpinner } from './spinner'
export { default as ajax } from './ajax'
export { default as redirectAuth } from './redirectAuth'
export { default as getOrderPayload } from './getOrderPayload'
export { default as eventEmitter } from './events'
export { default as getInstallmentOptions } from './getInstallmentOptions'
export { default as Checkout, createFromAction } from './checkout'
export { createPresentToShopperModal } from './modal'
export { default as createError } from './createError'
export { addDays } from './date'
