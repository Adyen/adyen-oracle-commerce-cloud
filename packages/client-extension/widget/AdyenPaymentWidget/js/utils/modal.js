import $ from 'jquery'
import hideLoaders from './hideLoaders'

export const createModal = () => {
    const modal = document.querySelector('#cc-editPane')
    if (!modal) {
        const node = `
        <!-- Edit Personalization Modal -->
        <div class="modal fade" id="cc-editPane" tabindex="-1" role="dialog">
            <div class="modal-dialog cc-modal-dialog">
                <div class="modal-content">
                    <div class="modal-body cc-modal-body">
                        <div id="present-shopper"></div>
                    </div>
                </div>
            </div>
        </div>
    `
        const template = document.createElement('template')
        template.innerHTML = node
        document.body.appendChild(template.content)
    }
}

export const showModal = () => {
    hideLoaders()
    $('#cc-editPane').modal('show')
}

export const hideModal = () => {
    $('#cc-editPane').modal('hide')
}

export const createPresentToShopperModal = (cb) => {
    const node = document.createElement('div')
    node.setAttribute(
        'style',
        'display: flex;justify-content: center;align-content: center;top: 0;position: fixed;z-index: 999;' +
            'align-items: center;'
    )

    const wrapper = document.createElement('div')
    wrapper.setAttribute('style', 'height: 100vh; width: 100vw; background-color: #a4a4a494; z-index: 999;')
    wrapper.setAttribute('id', 'present-shopper-wrapper')

    const modal = document.createElement('div')
    modal.setAttribute('id', 'present-shopper')
    modal.setAttribute('style', 'z-index: 9999;position: absolute;')

    const clickEvent = () => {
        wrapper.removeEventListener('click', clickEvent)
        node.remove()
    }

    wrapper.addEventListener('click', clickEvent)

    node.appendChild(modal)
    node.appendChild(wrapper)

    document.body.appendChild(node)
    cb(node)
}
