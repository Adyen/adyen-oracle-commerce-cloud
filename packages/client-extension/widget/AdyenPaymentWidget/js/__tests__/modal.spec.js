import { createPresentToShopperModal } from '../utils/modal'
import * as constants from '../constants'

describe('Modal', () => {
    it('should send node in the callback', function() {
        const spy = jest.fn()
        createPresentToShopperModal(spy)
        expect(spy.mock.calls[0][0]).toMatchSnapshot()
    })
})
