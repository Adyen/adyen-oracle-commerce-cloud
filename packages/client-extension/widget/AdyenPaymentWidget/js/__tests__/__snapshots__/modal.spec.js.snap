// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should send node in the callback 1`] = `
<div
  style="display: flex;justify-content: center;align-content: center;top: 0;position: fixed;z-index: 999;align-items: center;"
>
  <div
    id="present-shopper"
    style="z-index: 9999;position: absolute;"
  />
  <div
    id="present-shopper-wrapper"
    style="height: 100vh; width: 100vw; background-color: #a4a4a494; z-index: 999;"
  />
</div>
`;
